walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,63 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
+  /**
+   * Returns the times when {@code request} could happen on a day containing {@code events}
+   * Time complexity of the algorithm is O(nlogn + mn), in which
+   * n is the number of {@code events}, and m is the number of attendees in {@code request}
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
```

Comment:
Where is the validation for the `noOptionsForTooLongOfARequest` test?
I think the test passes because of the empty `events` being passed?

Seems also the same for the `optionsForNoAttendees` test.

Reply:
If the request duration is more than a day, it would not be possible to find a valid time to host the event, thus my algorithm would return an empty list.

When there are no attendees, the whole day would be returned.

I will add some code at the beginning of the method to handle invalid inputs and edge cases.